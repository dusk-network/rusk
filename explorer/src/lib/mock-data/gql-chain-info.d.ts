type GQLBlock = {
  fees: number;
  gasSpent: number;
  header: {
    gasLimit: number;
    hash: string;
    height: number;
    nextBlockHash?: string;
    prevBlockHash: string;
    seed: string;
    stateHash: string;
    timestamp: number;
  };
  reward: number;
  transactions: GQLTransaction[];
};

type GQLCallData = {
  contractId: string;
  fnName: string;
};

type GQLChainInfo = {
  blocks: GQLBlock[];
  transactions: GQLTransaction[];
};

type GQLSearchResult = {
  block?: {
    header: {
      hash: string;
    };
  } | null;
  tx?: {
    id: string;
  } | null;
  account?: {
    id: string;
  } | null;
};

type GQLTransaction = {
  blockHash: string;
  blockHeight: number;
  blockTimestamp: number;
  err: string | null;
  gasSpent: number;
  id: string;
  tx: {
    blobHashes: string[] | null;
    callData: GQLCallData | null;
    gasLimit: number;
    gasPrice: number;
    id: string;
    isDeploy: boolean;
    memo: string;
    txType: string;
    json?: string;
  };
};
