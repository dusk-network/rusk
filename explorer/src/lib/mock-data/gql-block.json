{
  "block": {
    "fees": 580718,
    "gasSpent": 580718,
    "header": {
      "gasLimit": 5000000000,
      "hash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
      "height": 495868,
      "nextBlockHash": "6011556208a85e6001bd01ccbf936486b91318a7f6cbcf7ab810adf6fae34204",
      "prevBlockHash": "07b74b35c2c7cf8f41426cd0870bafa1a2c7adee3fdd876643548096186fc4cb",
      "seed": "af15447e3a004a79d4ae8b084f7b76b78d95880bb63e1cfa79250a310731f52e6d84ee42a5d6fc2cb99c5b1f489761f6",
      "stateHash": "20bb0a677b93f084afadfd34bec3ac3feee33a020b81d9549afa2268e8543acb",
      "timestamp": 1713298022
    },
    "reward": 16000000000,
    "transactions": [
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 290866,
        "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "isDeploy": false,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
          "memo": "",
          "txType": "Moonlight"
        }
      },
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 289852,
        "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
          "isDeploy": false,
          "memo": "some notes about the transaction",
          "txType": "Phoenix"
        }
      }
    ]
  }
}
