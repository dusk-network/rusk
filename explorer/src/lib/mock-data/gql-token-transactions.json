[
  {
    "amount": 4359,
    "from": "0xf1c4230aca3ca19199d0f9665bbd9cff8",
    "to": "0x189f8d571b05adc6a35ef1bb0e3b458184",
    "blockhash": "103749d1e8990cdda301b2c4efd6c50735548dd2fff511d9d99c75",
    "blockheight": 6670,
    "date": "2025-02-04 21:15:22.373891+00:00",
    "feepaid": 364,
    "gaslimit": 40133828,
    "gasprice": 21,
    "gasspent": 8255387,
    "memo": "Mock transaction",
    "method": "stake",
    "success": true,
    "txerror": "Out of gas",
    "txid": "15fb0001ed66bb0dacc02ec003de09e31a3de10e810fa8bb384175",
    "txtype": "Phoenix",
    "payload": "0x",
    "nonce": 0
  }
]
