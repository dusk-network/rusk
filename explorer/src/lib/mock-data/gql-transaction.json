{
  "tx": {
    "blockHash": "3c6e4018cfa86723e50644e33d3990bc27fc794f6b49fbf6290e4d308e07bd2d",
    "blockHeight": 487166,
    "blockTimestamp": 1713249549,
    "gasSpent": 290766,
    "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
    "err": null,
    "tx": {
      "blobHashes": null,
      "callData": null,
      "gasLimit": 500000000,
      "gasPrice": 1,
      "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
      "isDeploy": false,
      "memo": "some notes about the transaction",
      "txType": "Moonlight",
      "json": "{\"type\":\"moonlight\",\"sender\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"receiver\":\"mHWZo9qYUhqp2SEmtWN7EuDKFwVrjgdEyzpHdfLB6R9scRAq5EkUQyLB9fBfCGt1wjxfEpFxHq9MBPVVPY3Lk3JKnQLWZzj7UYAR4mGmeGQZwCaeCS8uA63ZPKnGpjiUnj\",\"value\":9812378912731,\"nonce\":156,\"deposit\":0,\"fee\":{\"refund_address\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"gas_limit\":\"2500000\",\"gas_price\":\"1\"},\"call\":null,\"is_deploy\":false,\"memo\":null}"
    }
  }
}
