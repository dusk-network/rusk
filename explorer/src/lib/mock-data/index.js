export { default as apiMarketData } from "./api-market-data.json";
export { default as apiNodeInfo } from "./api-node-info.json";
export { default as apiNodeLocations } from "./api-node-locations.json";
export { default as apiStats } from "./api-stats.json";
export { default as gqlBlock } from "./gql-block.json";
export { default as gqlBlockDetails } from "./gql-block-details.json";
export { default as gqlBlocks } from "./gql-blocks.json";
export { default as gqlLatestChainInfo } from "./gql-latest-chain-info.json";
export { default as gqlSearchPossibleResults } from "./gql-search-possible-results.json";
export { default as gqlTransaction } from "./gql-transaction.json";
export { default as gqlTransactions } from "./gql-transactions.json";
export { default as hostProvisioners } from "./host-provisioners.json";
export { default as nodeLocationsCount } from "./node-locations.count.json";

export { default as accounts } from "./accounts.json";
export { default as tokens } from "./tokens.json";
export { default as gqlTokenTransactions } from "./gql-token-transactions.json";
