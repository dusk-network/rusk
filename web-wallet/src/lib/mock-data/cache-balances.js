const factor = BigInt(1e9);

export default [
  {
    address:
      "c3d4e5f6g7h8i9j0A1B2C3D4E5F6G7H8I9J0a1b2c3d4e5f6g7h8i9j0A1B2C3D4E5F6G7H8",
    balance: {
      publicBalance: {
        nonce: 1n,
        value: 567n * factor,
      },
      shieldedBalance: {
        spendable: 400n * factor,
        value: 1234n * factor,
      },
    },
  },
  {
    address:
      "B2C3D4E5F6G7H8I9J0a1b2c3d4e5f6g7h8i9j0A1B2C3D4E5F6G7H8I9J0a1b2c3d4e5f6g",
    balance: {
      publicPublic: {
        nonce: 5n,
        value: 123n * factor,
      },
      shieldedBalance: {
        spendable: 123n * factor,
        value: 456n * factor,
      },
    },
  },
];
