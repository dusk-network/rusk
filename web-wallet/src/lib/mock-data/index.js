export { default as addresses } from "./addresses";
export { default as cacheHistory } from "./cache-history";
export { default as cacheSpentNotes } from "./cache-spent-notes";
export { default as cacheUnspentNotes } from "./cache-unspent-notes";
export { default as stakeInfo } from "./stakeInfo";
export { default as transactions } from "./transactions";
