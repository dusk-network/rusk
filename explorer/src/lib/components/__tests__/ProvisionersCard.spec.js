import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { slice } from "lamb";

import { hostProvisioners } from "$lib/mock-data";
import { ProvisionersCard } from "..";

describe("Provisioners Card", () => {
  const data = slice(hostProvisioners, 0, 10);

  const baseProps = {
    error: null,
    isSmallScreen: false,
    loading: false,
    provisioners: null,
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("should render the `ProvisionersCard` component", () => {
    const { container } = render(ProvisionersCard, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button is the card is in the loading state", () => {
    const loading = true;

    const { container, getByRole } = render(ProvisionersCard, {
      ...baseOptions,
      props: { ...baseProps, loading },
    });

    expect(getByRole("button", { name: "Show More" })).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button if there is no more data to display", async () => {
    const loading = false;
    const provisioners = data;

    const { container, getByRole } = render(ProvisionersCard, {
      ...baseOptions,
      props: { ...baseProps, loading, provisioners },
    });

    expect(getByRole("button", { name: "Show More" })).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should hide the `Show More` button if an error has occurred", async () => {
    const props = { ...baseProps, error: new Error("error") };

    const { container } = render(ProvisionersCard, {
      ...baseOptions,
      props: { ...props, error: new Error("error") },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
