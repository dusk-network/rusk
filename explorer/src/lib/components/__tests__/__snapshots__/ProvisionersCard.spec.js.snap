// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Provisioners Card > should disable the \`Show More\` button if there is no more data to display 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Provisioners â€” 10 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    class="data-card__content"
  >
    <div
      class="table-container provisioners-table data-card__table"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="table__row"
          >
            <th
              class="table__header-cell align-left"
            >
              Staking Address
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Owner
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Stake
            </th>
            
             
            <th
              class="table__header-cell align-left"
            >
              Slashes
            </th>
            
             
            <th
              class="table__header-cell align-right"
            >
              Accumulated Reward
            </th>
            
          </tr>
        </thead>
         
        <tbody
          class="table__body"
        >
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              m6dy2gz3jC...tW82FWtnpf
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="m6dy2gz3jC...tW82FWtnpf"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="7,121.257296087 DUSK"
                data-tooltip-type="info"
              >
                7,121
                .
                <span
                  class="decimal-shadow"
                >
                  26
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              m9dVuRgr3C...94r33pFQyz
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="m9dVuRgr3C...94r33pFQyz"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="6,923.840604194 DUSK"
                data-tooltip-type="info"
              >
                6,923
                .
                <span
                  class="decimal-shadow"
                >
                  84
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mHGvQ9Xdjz...iTfxTvgn7f
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="0200000000000000000000000000000000000000000000000000000000000000"
              >
                Contract
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              4,800
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              4,320
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="89.505343279 DUSK"
                data-tooltip-type="info"
              >
                89
                .
                <span
                  class="decimal-shadow"
                >
                  51
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mLx5HUo5Ph...BrRgEzoka8
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="mLx5HUo5Ph...BrRgEzoka8"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              1,234
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              4,320
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="29.633646022 DUSK"
                data-tooltip-type="info"
              >
                29
                .
                <span
                  class="decimal-shadow"
                >
                  63
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mTgXDqkyVa...gAseybAsfU
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="0200000000000000000000000000000000000000000000000000000000000000"
              >
                Contract
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="7,518.756679536 DUSK"
                data-tooltip-type="info"
              >
                7,518
                .
                <span
                  class="decimal-shadow"
                >
                  76
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mUJHMDEBiT...FHNxRNhoAA
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="0200000000000000000000000000000000000000000000000000000000000000"
              >
                Contract
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              4,800
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              23,760
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="73.800637305 DUSK"
                data-tooltip-type="info"
              >
                73
                .
                <span
                  class="decimal-shadow"
                >
                  80
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mUUpjnXow2...SfzQcywzaW
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="0200000000000000000000000000000000000000000000000000000000000000"
              >
                Contract
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="7,653.541671166 DUSK"
                data-tooltip-type="info"
              >
                7,653
                .
                <span
                  class="decimal-shadow"
                >
                  54
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mWvq9EWD9b...oU7BY2oEDe
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="mWvq9EWD9b...oU7BY2oEDe"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              1,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              8,640
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="1.613840782 DUSK"
                data-tooltip-type="info"
              >
                1
                .
                <span
                  class="decimal-shadow"
                >
                  61
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mXKJ97xTtr...JgkCK2o7x4
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="mXKJ97xTtr...JgkCK2o7x4"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="7,512.060104659 DUSK"
                data-tooltip-type="info"
              >
                7,512
                .
                <span
                  class="decimal-shadow"
                >
                  06
                </span>
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell align-left"
            >
              mdXsA3Ee1Y...K8VZPimZ2J
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral"
                data-tooltip-id="provisioners-tooltip"
                data-tooltip-text="mdXsA3Ee1Y...K8VZPimZ2J"
              >
                Account
              </span>
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__stake-data-label"
              >
                Active:
              </b>
               
              500,000
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Inactive:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__stake-data-label"
              >
                Maturity At:
              </b>
              
          #
              0
            </td>
            
             
            <td
              class="table__data-cell align-left"
            >
              <b
                class="provisioners-table__slash-data-label"
              >
                Soft:
              </b>
               
              0
               
              <br />
               
              <b
                class="provisioners-table__slash-data-label"
              >
                Hard:
              </b>
               
              0
            </td>
            
             
            <td
              class="table__data-cell align-right"
            >
              <span
                data-tooltip-id="main-tooltip"
                data-tooltip-place="top"
                data-tooltip-text="7,973.118109253 DUSK"
                data-tooltip-type="info"
              >
                7,973
                .
                <span
                  class="decimal-shadow"
                >
                  12
                </span>
              </span>
            </td>
            
             
          </tr>
          
        </tbody>
      </table>
    </div>
    
  </div>
  
   
</div>
`;

exports[`Provisioners Card > should disable the \`Show More\` button is the card is in the loading state 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Provisioners â€” 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    aria-label="Loading"
    class="dusk-progress-bar data-card__progress-bar"
    role="progressbar"
  >
    <div
      class="dusk-progress-bar__filler dusk-progress-bar__filler--undetermined"
    />
  </div>
  
   
</div>
`;

exports[`Provisioners Card > should hide the \`Show More\` button if an error has occurred 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Provisioners â€” 0 Displayed Items
    </h1>
     
  </header>
   
  <div
    class="alert"
  >
    <svg
      class="dusk-icon dusk-icon--size--large"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,2L1,21H23M12,6L19.53,19H4.47M11,10V14H13V10M11,16V18H13V16"
      />
    </svg>
    
     
    <header>
      <p>
        There was an error fetching the data.
      </p>
       
      <p
        class="alert__error-message"
      >
        error
      </p>
    </header>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--secondary dusk-button--size--normal"
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Retry
      </span>
    </button>
  </div>
  
   
</div>
`;

exports[`Provisioners Card > should render the \`ProvisionersCard\` component 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Provisioners â€” 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  
   
</div>
`;
